packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -875,6 +931,93 @@ export class IntervalCollection<TInterval extends ISerializableInterval> extends
         return interval;
     }
 
+    public change(id, start: number, end: number, props?: MergeTree.PropertySet): TInterval | undefined {
+        if (!this.attached) {
+            throw new Error("Attach must be called before accessing intervals");
+        }
+        const interval = this.localCollection.changeIntervalById(id, start, end, props);
+        if (interval) {
+            const serializedInterval: ISerializedInterval = interval.serialize(this.client);
+            this.emitter.emit("change", undefined, serializedInterval);
+            this.addPendingChange(serializedInterval);
+        }
+        this.emit("change", interval, true, undefined);
+        return interval;
+    }
+
+    private addPendingChange(serializedInterval: ISerializedInterval) {
+        const id = serializedInterval.properties?.[reservedIntervalIdKey];
+        if (id === undefined) {
+            return;
+        }
+        if (!this.pendingChanges) {
+            this.pendingChanges = new Map<string, ISerializedInterval[]>();
```

Comment:
i don't think this is enough meta data, start, end, and properties should be individually managed